src/plugins/analysis/cwe_checker/code/cwe_checker.py

```diff
@@ -106,38 +65,36 @@ def _parse_module_versions(bap_output):
         module_versions = {}
         for line in bap_output.splitlines():
             if 'module_versions:' in line:
-                version_sexp = line.split('module_versions:')[-1].strip()
-                module_versions = dict(sexpdata.loads(version_sexp))
+                version_json = line.split('module_versions:')[-1].strip()
+                module_versions = json.loads(version_json)
         return module_versions
 
     @staticmethod
     def _build_bap_command_for_modules_versions():
         # unfortunately, there must be a dummy file passed to BAP, I chose /bin/true because it is damn small
-        return 'docker run --rm {} bap /bin/true --pass=cwe-checker --cwe-checker-module_versions=true'.format(DOCKER_IMAGE)
+        return 'docker run --rm {} bap /bin/true --pass=cwe-checker --cwe-checker-module-versions'.format(DOCKER_IMAGE)
 
     @staticmethod
     def _build_bap_command(file_object):
         return 'timeout --signal=SIGKILL {}m docker run --rm -v {}:/tmp/input {} bap /tmp/input '\
-               '--pass=cwe-checker --cwe-checker-config=/home/bap/cwe_checker/src/config.json'.format(BAP_TIMEOUT, file_object.file_path, DOCKER_IMAGE)
+               '--pass=cwe-checker --cwe-checker-json --cwe-checker-no-logging'.format(BAP_TIMEOUT, file_object.file_path, DOCKER_IMAGE)
 
     @staticmethod
     def _parse_bap_output(output):
         tmp = defaultdict(list)
-        cwe_parser = CweWarningParser()
-
-        for line in output.splitlines():
-            if 'WARN' in line:
-                cwe_warning = cwe_parser.parse(line)
-                tmp[cwe_warning.name].append(cwe_warning)
+        j = json.loads(output)
+        if 'warnings' in j:
+            for warning in j['warnings']:
+                tmp[warning['name']] = tmp[warning['name']] + [warning]
```

Comment:
I'd prefer the [warning, ] syntax to explicitly show warning to be one element. If warning would be a string, the string would be split into chars using the syntax without trailing comma